examples/with-redux-observable/pages/index.js

```diff
@@ -1,49 +1,48 @@
-import { Component } from 'react'
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
 import Link from 'next/link'
 import { of, Subject } from 'rxjs'
 import { StateObservable } from 'redux-observable'
-import { connect } from 'react-redux'
-import CharacterInfo from '../components/CharacterInfo'
-import { rootEpic } from '../redux/epics'
-import * as actions from '../redux/actions'
+import UserInfo from '../components/UserInfo'
+import { rootEpic } from '../store/epics'
+import {
+  stopFetchingUsers,
+  startFetchingUsers,
+  fetchUser,
+} from '../store/actions'
+import { initializeStore } from '../store/store'
 
-class Counter extends Component {
-  static async getInitialProps({ store, isServer }) {
-    const state$ = new StateObservable(new Subject(), store.getState())
-    const resultAction = await rootEpic(
-      of(actions.fetchCharacter(isServer)),
-      state$
-    ).toPromise() // we need to convert Observable to Promise
-    store.dispatch(resultAction)
+const Counter = () => {
+  const dispatch = useDispatch()
 
-    return { isServer }
-  }
+  useEffect(() => {
+    dispatch(startFetchingUsers())
+    return () => {
+      dispatch(stopFetchingUsers())
+    }
+  }, [dispatch])
 
-  componentDidMount() {
-    this.props.startFetchingCharacters()
-  }
+  return (
+    <div>
+      <h1>Index Page</h1>
+      <UserInfo />
+      <br />
+      <nav>
+        <Link href="/other">
+          <a>Navigate to "/other"</a>
+        </Link>
+      </nav>
+    </div>
+  )
+}
 
-  componentWillUnmount() {
-    this.props.stopFetchingCharacters()
-  }
+export async function getStaticProps() {
+  const store = initializeStore()
+  const state$ = new StateObservable(new Subject(), store.getState())
+  const resultAction = await rootEpic(of(fetchUser()), state$).toPromise() // we need to convert Observable to Promise
+  store.dispatch(resultAction)
 
-  render() {
-    return (
-      <div>
-        <h1>Index Page</h1>
-        <CharacterInfo />
-        <br />
-        <nav>
-          <Link href="/other">
-            <a>Navigate to "/other"</a>
-          </Link>
-        </nav>
-      </div>
-    )
-  }
+  return { props: {} }
```

Comment:
If the returned props are empty, then this `getStaticProps` is not doing anything, you could update it or just remove it entirely. Ideally remove it.

Reply:
Resolved.